### PR TITLE
Completed changelog

### PR DESCRIPTION
**Pull request by zahidalidev**
Before you merge this PR, please review the following:

## Blog post
- [ ] Checked title case (This is correct. This Is Not Correct)
- [ ] Updated publishDate
- [ ] Updated authors
- [ ] Updated and checked slug
- [ ] Proofread for spelling and grammar errors
- [ ] Checked for consistent tone and voice throughout via Grammarly
- [ ] Ensured all links are working correctly
- [ ] Added relevant tags and categories
- [ ] Optimized meta description
- [ ] Included a new hero and social/OG image
- [ ] Uploaded images to Google Storage
- [ ] Formatted text for readability (headings, bullet points, etc.)
- [ ] Added image alt text
- [ ] Ran [Markdown linter](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint)

## Other

- [x] This is not a blog post

### CODE REVIEW

**Comment by MaedahBatool** [comment]
@zahidalidev let's keep this as a draft PR until we address all the issues. I have also asked the design team to review everything.

**Comment by zahidalidev** [comment]
Sure, got it.

File: src/hooks/loadMoreAndSearch.ts
Comment (traceyljohnson) [inline src/hooks/loadMoreAndSearch.ts:1]: Can we update the search input box to use the focus color shown in the mockup (`violet-500`) when user is typing?

**Review by traceyljohnson — COMMENTED** [review]
Left comments with some design feedback! cc @MaedahBatool

**Comment by MaedahBatool** [comment]
Thanks @traceyljohnson for the review. I'll look into it. :)

File: public/assets/changelog/bento-hero.png
[the remark this answers is not in the document]
Comment (MaedahBatool) [reply public/assets/changelog/bento-hero.png:1]: Sure @zahidalidev let's try to make it an SVG.

File: src/components/Changelog/PostDetailsInfo.tsx
[the remark this answers is not in the document]
Comment (zahidalidev) [reply src/components/Changelog/PostDetailsInfo.tsx:1]: Done.

File: src/components/Changelog/icons/ChevronDownIcon.tsx
[the remark this answers is not in the document]
Comment (zahidalidev) [reply src/components/Changelog/icons/ChevronDownIcon.tsx:1]: Done.

File: tailwind.config.js [hunk 1]
```diff
@@ -149,6 +181,14 @@ module.exports = {
        * See: https://tailwindcss.com/docs/responsive-design
        * sm: 640px, md: 768px, lg: 1024px, xl: 1280px, 2xl: 1536px
        */
+      lineHeight: {
+        'extra-tight': '57.6px',
+        'custom-150': '150%',
+      },
+      letterSpacing: {
```
[the remark this answers is not in the document]
Comment (zahidalidev) [reply tailwind.config.js:188]: Yes, I've removed them.

File: src/pages/changelog/index.tsx [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+
+    return (
+        <Layout className="bg-[#F9FAFB]">
+            <div className="relative mt-[-30px] flex items-center justify-center bg-[#F9FAFB]">
+                <div
+                    className="relative flex h-[330px] w-[85%] items-center justify-center rounded-lg bg-cover bg-center"
+                    // eslint-disable-next-line react/forbid-dom-props
+                    style={{ backgroundImage: 'url("/assets/changelog/bento-hero.png")' }}
+                >
+                    <div className="absolute inset-0 flex flex-col items-start justify-center p-6 px-[80px]">
```
[the remark this answers is not in the document]
Comment (zahidalidev) [reply src/pages/changelog/index.tsx:89]: Done.

File: src/hooks/loadMoreAndSearch.ts
Comment (zahidalidev) [reply src/hooks/loadMoreAndSearch.ts:1]: Done.

File: src/pages/changelog/index.tsx [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+        1,
+        posts
+    )
+
+    const allTags = currentRecords
+        .map(record => record.frontmatter.tags)
+        .flat()
+        .filter((tag, index, self) => self.indexOf(tag) === index)
+
+    const secondaryTags = [...allTags.filter(tag => (tag as string) && !primaryTopics.includes(tag as string) && !secondaryTopics.includes(tag as string)), ...secondaryTopics].sort()
```
[the remark this answers is not in the document]
Comment (MaedahBatool) [reply src/pages/changelog/index.tsx:32]: @traceyljohnson We wanted to show important ones always above the fold, like Cody, VS Code, and JetBrains should always appear. So we cannot make it alphabetical.

File: src/components/Changelog/PreCodeBlock.tsx
[the remark this answers is not in the document]
Comment (zahidalidev) [reply src/components/Changelog/PreCodeBlock.tsx:1]: @traceyljohnson can you please point me to `code-sm` as I cannot find it in `src/styles/globals.css`.

File: src/components/Changelog/icons/SearchIcon.tsx
[the remark this answers is not in the document]
Comment (MaedahBatool) [reply src/components/Changelog/icons/SearchIcon.tsx:1]: @zahidalidev is this done? If yes, let's mark it as complete.

File: src/components/Changelog/icons/TagIcon.tsx
[the remark this answers is not in the document]
Comment (MaedahBatool) [reply src/components/Changelog/icons/TagIcon.tsx:1]: @zahidalidev is this done? If yes, let's mark it as complete.

File: tailwind.config.js [hunk 4]
```diff
@@ -149,6 +181,14 @@ module.exports = {
        * See: https://tailwindcss.com/docs/responsive-design
        * sm: 640px, md: 768px, lg: 1024px, xl: 1280px, 2xl: 1536px
        */
+      lineHeight: {
```
[the remark this answers is not in the document]
Comment (MaedahBatool) [reply tailwind.config.js:184]: @zahidalidev is this done? If yes, let's mark it as complete.

File: public/assets/changelog/bento-hero.png
[the remark this answers is not in the document]
Comment (zahidalidev) [reply public/assets/changelog/bento-hero.png:1]: Completed.

**Comment by kukicado** [comment]
LGTM!

A few small improvements:

- Can we make the version a URL param like the topics? This way, we can link to version specific updates.
- For the subheading: Latest updates to the Sourcegraph and Cody platform, can we change this to: "Latest updates to the Sourcegraph Code Search and Cody platforms"
- For individual updates, can we have the styles match the blog (with bigger text, wider width, and clickable screenshots that expand to full view) 
CleanShot 2024-08-27 at 09 54 26@2x
- For individual updates, can we have a back to Changelog button like we do on the blog
- For individual updates, can we have the topic/versions clickable that go back to /changelog?{params}
- For individual updates, can we maybe show the tags on a new line, or collapse them if there's multiple. This for example seems a bit messy:
CleanShot 2024-08-27 at 09 57 16@2x

**Comment by jdorfman** [comment]
@zahidalidev @MaedahBatool make sure to update the `Changelog` link in the footer to `/changelog`

**Comment by MaedahBatool** [comment]
Hey everyone, thank you for reviewing the PR. We have addressed all the critical feedback. All the remaining items will be tackled in the next iteration, which I will schedule in another few weeks. I appreciate all the patience here. Let's ship it 🚀

**Review by MaedahBatool — APPROVED** [review]
LGTM let's go 🚀

**Pull request merged**